cuqi/experimental/mcmc/_gibbs.py

```diff
@@ -0,0 +1,217 @@
+from cuqi.distribution import JointDistribution
+from cuqi.experimental.mcmc import SamplerNew
+from cuqi.samples import Samples
+from typing import Dict
+import numpy as np
+import warnings
+
+try:
+    from progressbar import progressbar
+except ImportError:
+    def progressbar(iterable, **kwargs):
+        warnings.warn("Module mcmc: Progressbar not found. Install progressbar2 to get sampling progress.")
+        return iterable
+
+# Not subclassed from SamplerNew as Gibbs handles multiple samplers and samples multiple parameters
+# Similar approach as for JointDistribution
+class GibbsNew: 
+    """
+    Gibbs sampler for sampling a joint distribution.
+
+    Gibbs sampling samples the variables of the distribution sequentially,
+    one variable at a time. When a variable represents a random vector, the
+    whole vector is sampled simultaneously.
+    
+    The sampling of each variable is done by sampling from the conditional
+    distribution of that variable given the values of the other variables.
+    This is often a very efficient way of sampling from a joint distribution
+    if the conditional distributions are easy to sample from.
+
+    In each Gibbs step, the corresponding sampler has the initial_point set
+    to the previous sample and the sampler is reinitialized.
+
```

Comment:
Are there any limitations on the current Gibbs that should be mentioned? For example does it only work for conjugate pairs or how general distributions can one specify?